exec/java-exec/src/main/java/org/apache/drill/exec/store/httpd/HttpdLogRecord.java

```diff
@@ -20,31 +20,40 @@
 import org.apache.drill.shaded.guava.com.google.common.base.Charsets;
 import org.apache.drill.shaded.guava.com.google.common.collect.Maps;
 import io.netty.buffer.DrillBuf;
+
 import java.util.EnumSet;
 import java.util.Map;
+
 import nl.basjes.parse.core.Casts;
 import nl.basjes.parse.core.Parser;
 import org.apache.drill.exec.vector.complex.writer.BaseWriter.MapWriter;
 import org.apache.drill.exec.vector.complex.writer.BigIntWriter;
 import org.apache.drill.exec.vector.complex.writer.Float8Writer;
 import org.apache.drill.exec.vector.complex.writer.VarCharWriter;
+import org.apache.drill.exec.vector.complex.writer.TimeStampWriter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class HttpdLogRecord {
 
   private static final Logger LOG = LoggerFactory.getLogger(HttpdLogRecord.class);
   private final Map<String, VarCharWriter> strings = Maps.newHashMap();
   private final Map<String, BigIntWriter> longs = Maps.newHashMap();
   private final Map<String, Float8Writer> doubles = Maps.newHashMap();
+  private final Map<String, TimeStampWriter> times = Maps.newHashMap();
   private final Map<String, MapWriter> wildcards = Maps.newHashMap();
   private final Map<String, String> cleanExtensions = Maps.newHashMap();
   private final Map<String, MapWriter> startedWildcards = Maps.newHashMap();
   private final Map<String, MapWriter> wildcardWriters = Maps.newHashMap();
   private DrillBuf managedBuffer;
+  private String timeFormat;
 
-  public HttpdLogRecord(final DrillBuf managedBuffer) {
+  public HttpdLogRecord(final DrillBuf managedBuffer, String timeFormat) {
```

Comment:
It looks as if `timeFormat`could be final. Or even instead of `timeFormat` field there could be `private final SimpleDateFormat dateFormatter` one (which is created in constructor) to avoid its creation in `setTimeStamp` method every time, what do you think?

Reply:
Fixed.  Good call.  I moved it to the constructor so that it only happens once.